javagen/src/main/java/com/azure/autorest/template/example/ClientMethodExampleWriter.java

```diff
@@ -57,6 +66,106 @@ public ClientMethodExampleWriter(ClientMethod method, String clientVarName, Prox
+        if (isGroupingParameter(convenienceMethod, methodParameter)) {
+            // grouping, possible with flattening first
+
+            // group example values into a map
+            Map<String, Object> exampleValue = new HashMap<>();
+            for (MethodTransformationDetail detail : convenienceMethod.getMethodTransformationDetails()) {
+                for (ParameterMapping parameterMapping : detail.getParameterMappings()) {
+                    String serializedParameterName = parameterMapping.getInputParameterProperty().getSerializedName();
+                    ClientMethodParameter parameter = detail.getOutParameter();
+                    if (parameterMapping.getOutputParameterPropertyName() != null) {
+                        // this is a flattened property, so put flattening(real parameter) value
+                        serializedParameterName = detail.getOutParameter().getName();
+                        Map<String, Object> flattenedParameterValue = (Map<String, Object>) ModelExampleUtil.getParameterExampleValue(proxyMethodExample, serializedParameterName, parameter.getRequestParameterLocation());
+                        if (flattenedParameterValue != null) {
+                            exampleValue.putAll(flattenedParameterValue);
+                        }
+                    } else {
+                        exampleValue.put(serializedParameterName,
+                                ModelExampleUtil.getParameterExampleValue(
+                                        proxyMethodExample, serializedParameterName, parameter.getRequestParameterLocation()));
+                    }
+                }
+            }
+            IType type = methodParameter.getClientMethodParameter().getClientType();
+            IType wireType = methodParameter.getClientMethodParameter().getWireType();
+            return ModelExampleUtil.parseNode(type, wireType, exampleValue);
+        } else if (isFlattenParameter(convenienceMethod, methodParameter)) {
+            // flatten, no grouping
+            String realParameterName = convenienceMethod.getMethodTransformationDetails().iterator().next().getOutParameter().getName();
+            Map<String, Object> realParameterValue = (Map<String, Object>) proxyMethodExample.getParameters().get(realParameterName).getObjectValue();
+
+            IType type = methodParameter.getClientMethodParameter().getClientType();
+            IType wireType = methodParameter.getClientMethodParameter().getWireType();
+
+            Object flatteningParameterValue = null;
+            if (realParameterValue != null) {
+                flatteningParameterValue = realParameterValue
+                        .entrySet()
+                        .stream()
+                        .filter(entry -> Objects.equals(
+                                CodeNamer.getEscapedReservedClientMethodParameterName(entry.getKey()),
+                                methodParameter.getClientMethodParameter().getName()))
```

Comment:
I am not sure about this part. Should the comparison be based on "serialized name"?

And the logic here is different from the "flatten" process in grouping?

Reply:
Changed the check the outputParameterProperty's serializedName.